### PR TITLE
Binary Search

### PR DESCRIPTION
**Pull request by qjd2413**

### CODE REVIEW

File: exercises/binary-search/binary_search_tests.plt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+:- begin_tests(binary_search).
+
+    test(finds_a_value_in_an_array_with_one_element) :-
```
Comment (parkerl) [inline exercises/binary-search/binary_search_tests.plt]: This always gets me when switching languages. We don't use "arrays" in Prolog. Change it to "list" in the test names please?

File: exercises/binary-search/binary_search.pl [hunk 2]
```diff
@@ -0,0 +1 @@
+binary_search(Lst, Val, Index).
```
Comment (parkerl) [inline exercises/binary-search/binary_search.pl]: Can we fully spell out the variables? (List, Value, Index)

**Comment by parkerl** [comment]
I'm afraid this has the same issue that was raised here https://github.com/exercism/xelixir/issues/176

You did a nice job on this but we need to make sure that it is appropriate to have this exercise in prolog.

**Comment by qjd2413** [comment]
You're right, this implementation is best case O(N), which makes the binary search useless. I didnt consider how it would perform in this specific language. Should this PR be closed?

**Comment by parkerl** [comment]
Yes. I think we should close. Sorry.

**Pull request closed without merging**